components/button/index.js

```diff
@@ -4,22 +4,43 @@
 import './style.scss';
 import classnames from 'classnames';
 
-function Button( { href, target, isPrimary, isLarge, isToggled, className, disabled, ...additionalProps } ) {
-	const classes = classnames( 'components-button', className, {
-		button: ( isPrimary || isLarge ),
-		'button-primary': isPrimary,
-		'button-large': isLarge,
-		'is-toggled': isToggled,
-	} );
-
-	const tag = href !== undefined && ! disabled ? 'a' : 'button';
-	const tagProps = tag === 'a' ? { href, target } : { type: 'button', disabled };
-
-	return wp.element.createElement( tag, {
-		...tagProps,
-		...additionalProps,
-		className: classes,
-	} );
+class Button extends wp.element.Component {
+	constructor( props ) {
+		super( props );
+		this.setRef = this.setRef.bind( this );
+	}
+
+	componentDidMount() {
+		if ( this.props.focus ) {
```

Comment:
>Could you explain the issues with focus you were seeing? Do we need to monitor this prop if it changes during the lifetime of the component?

Circling back to this. I encountered the Toolbar `focus` prop when working on some unrelated refactoring and had _no_ idea what its purpose is. I still don't, and could use some clarification as to this unanswered question.